File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::map<int, int> m;
        for (int num : nums) {
            if (m[num] >= 1) return true;
            m[num]++;
        }
        return false;
    }
};

/*
*/

File: Data Structures & Algorithms/split-array-largest-sum/submission-0.py
class Solution:
    def splitArray(self, nums: List[int], k: int) -> int:
        
        def splittable(max_sum):
            running_sum = 0
            buckets = 1
            for num in nums:
                if running_sum + num > max_sum:
                    running_sum = 0
                    buckets += 1
                running_sum += num
            return buckets

        l, r = max(nums), sum(nums)

        while l <= r:
            mid = (l + r) // 2
            if splittable(mid) <= k:
                r = mid - 1
            else:
                l = mid + 1
        
        return l
